### PR TITLE
Feature/edit task

### PR DESCRIPTION
**Pull request by nhuquyenit99**

### CODE REVIEW

File: public/js/main.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ function saveData(todoList){
 }
 
 function convertToHTML(list) {
-  var content = list.map(function (item) {
+  let content = list.map(function (item) {
```
Comment (hwth) [inline public/js/main.js:20]: please use `const` for unchanged variables  

- [ ] fix all the same issue

File: public/js/main.js [hunk 2]
```diff
@@ -157,25 +157,25 @@ btnToggleAll.addEventListener("click", () => {
     });
   }
   saveData(todoList);
-  renderData();
+  todoListComputed();
 });
 
-function renderData() {
+function todoListComputed() {
```
Comment (hwth) [inline public/js/main.js:163]: A method should start with a verb

- [ ] refactor

File: public/js/main.js [hunk 3]
```diff
@@ -185,9 +185,34 @@ filters.addEventListener("click",(event) => {
   const filterSelected = document.getElementsByClassName("selected")[0];
   filterSelected.removeAttribute("class");
   element.className = "selected";
-  renderData();
+  todoListComputed();
 });
 
+listData.addEventListener("dblclick",(event) => {
+  const element = event.target;
+  const item = element.parentNode.parentNode;
```
Comment (hwth) [inline public/js/main.js:193]: A variable should start with their scope or object are pointing. This case it's task element.
E.g: `item` -> `taskElement`
- [ ] rename

File: public/js/main.js [hunk 4]
```diff
@@ -185,9 +185,34 @@ filters.addEventListener("click",(event) => {
...
+  item.classList.add("editing");
+  const inputEdit = item.lastElementChild;
+
+  inputEdit.onkeypress = (e) => {
```
Comment (hwth) [inline public/js/main.js:197]: Why do we bind the event here?

- [ ] explain

File: public/js/main.js [hunk 5]
```diff
@@ -185,9 +185,34 @@ filters.addEventListener("click",(event) => {
...
+    const enterKey = 13;
+    if (e.charCode === enterKey && inputEdit.value.trim() !== "") {
+      const taskId = item.getAttribute("id");
+      todoList.map(item => {
```
Comment (hoaibkdn) [inline public/js/main.js:201]: `map` will return an array data, however, you didn't assign to any variable, just use it as a loop, should replace to `foreach`

File: public/js/main.js [hunk 6]
```diff
@@ -185,9 +185,34 @@ filters.addEventListener("click",(event) => {
...
 });
 
+listData.addEventListener("dblclick",(event) => {
+  const element = event.target;
+  const item = element.parentNode.parentNode;
+  item.classList.add("editing");
+  const inputEdit = item.lastElementChild;
+
+  inputEdit.onkeypress = (e) => {
+    const enterKey = 13;
```
Comment (hoaibkdn) [inline public/js/main.js:198]: `enterKey` variable duplicated in many functions, should put it out of the function
and should capitalize this one

File: public/js/main.js (hunk 4 above)
Comment (nhuquyenit99) [reply public/js/main.js:198]: I want to get the key pressed and then compare it with "enterKey" to know when user presses enter

Comment (hwth) [reply public/js/main.js:198]: The event is bound every time the `dblclick` is fired. Can we bind this event when the input is initialized?

- [ ] consider

File: public/js/main.js [hunk 7]
```diff
@@ -141,11 +142,11 @@ listData.addEventListener("click", (event) => {
 });
 
 btnToggleAll.addEventListener("click", () => {
-  let count = todoList.reduce((count, item) => {
+  let countCompletedTasks = todoList.reduce((count, item) => {
```
Comment (hwth) [inline public/js/main.js:145]: The variable should be a noun. Rename to `completedTasksCounter`

- [ ] rename

**Pull request merged**